src/aks-preview/azext_aks_preview/addonconfiguration.py

```diff
@@ -0,0 +1,900 @@
+    if need_post_creation_role_assignment:
+        # adding a wait here since we rely on the result for role assignment
+        result = LongRunningOperation(cmd.cli_ctx)(
+            client.begin_create_or_update(resource_group_name, name, instance))
+        cloud_name = cmd.cli_ctx.cloud.name
+        # mdm metrics supported only in Azure Public cloud so add the role assignment only in this cloud
+        if monitoring_addon_enabled and cloud_name.lower() == 'azurecloud':
+            from msrestazure.tools import resource_id
+            cluster_resource_id = resource_id(
+                subscription=subscription_id,
+                resource_group=resource_group_name,
+                namespace='Microsoft.ContainerService', type='managedClusters',
+                name=name
+            )
+            add_monitoring_role_assignment(result, cluster_resource_id, cmd)
+        if ingress_appgw_addon_enabled:
+            add_ingress_appgw_addon_role_assignment(result, cmd)
+        if enable_virtual_node:
+            # All agent pool will reside in the same vnet, we will grant vnet level Contributor role
+            # in later function, so using a random agent pool here is OK
+            random_agent_pool = result.agent_pool_profiles[0]
+            if random_agent_pool.vnet_subnet_id != "":
+                add_virtual_node_role_assignment(
+                    cmd, result, random_agent_pool.vnet_subnet_id)
+            # Else, the cluster is not using custom VNet, the permission is already granted in AKS RP,
+            # we don't need to handle it in client side in this case.
+
+    else:
+        result = sdk_no_wait(no_wait, client.begin_create_or_update,
+                             resource_group_name, name, instance)
+    return result
+
+
+def update_addons(cmd,  # pylint: disable=too-many-branches,too-many-statements
+                  instance,
+                  subscription_id,
+                  resource_group_name,
+                  name,
+                  addons,
+                  enable,
+                  check_enabled=True,
+                  workspace_resource_id=None,
+                  enable_msi_auth_for_monitoring=False,
+                  subnet_name=None,
+                  appgw_name=None,
+                  appgw_subnet_prefix=None,
+                  appgw_subnet_cidr=None,
+                  appgw_id=None,
+                  appgw_subnet_id=None,
+                  appgw_watch_namespace=None,
+                  enable_sgxquotehelper=False,
+                  enable_secret_rotation=False,
+                  no_wait=False):  # pylint: disable=unused-argument
+    # parse the comma-separated addons argument
+    addon_args = addons.split(',')
+
+    addon_profiles = instance.addon_profiles or {}
+
+    os_type = 'Linux'
+
+    # for each addons argument
+    for addon_arg in addon_args:
+        if addon_arg not in ADDONS:
+            raise CLIError("Invalid addon name: {}.".format(addon_arg))

```

Comment:
Could you please replace all `CLIError` with more specific error types defined in `azure.cli.core.azclierror`? Also don’t forget to change the corresponding error type checked in test cases.

Reply:
This code was not written by me. I just moved it to organize everything. Is it ok for me to still change the errors?